### PR TITLE
#594 Implement clustering procedure for given list of assay values

### PR DESCRIPTION
**Pull request by suppnova**

### CODE REVIEW

File: api/python/setup.py [hunk 1]
```diff
@@ -123,4 +123,5 @@
     },
     download_url="https://pypi.org/project/epam.indigo",
     test_suite="tests",
+    extras_require={"ml": ["scikit-learn", "pytest"]},
```
Comment (mkviatkovskii) [inline api/python/setup.py:126]: We don't need pytest here, since it's test dependency, not runtime.

File: api/python/indigo/ml/clustering.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
+from sklearn.cluster import SpectralClustering
+
+
+def clustering(
+    assay_values: list[list], method=SpectralClustering, **kwargs
+) -> list:
```
Comment (mkviatkovskii) [inline api/python/indigo/ml/clustering.py:6]: Please specify output list type, I guess it should be `list[int]` in this case.

File: api/python/indigo/ml/clustering.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
+from sklearn.cluster import SpectralClustering
+
+
+def clustering(
+    assay_values: list[list], method=SpectralClustering, **kwargs
```
Comment (mkviatkovskii) [inline api/python/indigo/ml/clustering.py:5]: Please specify list type, maybe like `list[list[Union[int, float]]]`?

Also `method: Type[ClusterMixing]` seems to be reasonable.

**Review by mkviatkovskii — CHANGES_REQUESTED** [review]
Looks good, just need to fix couple cosmetic issues.

File: api/python/setup.py (hunk 1 above)
Comment (suppnova) [reply api/python/setup.py]: Pytest moved to requirements_dev.txt

File: api/python/indigo/ml/clustering.py (hunk 2 above)
Comment (suppnova) [reply api/python/indigo/ml/clustering.py]: Done

File: api/python/indigo/ml/clustering.py (hunk 3 above)
Comment (suppnova) [reply api/python/indigo/ml/clustering.py]: Done. 
But ClusterMixing caused some IDE highlights. Still thinking about collecting all valid clustering methods into a list for type checking

**Review by mkviatkovskii — APPROVED** [review]

**Pull request merged**